### PR TITLE
Updated prombenchy to support multiple benchmarks on one cluster; adjusted avalanche, removed some experiments for now.

### PR DESCRIPTION
**Pull request by bwplotka**
Good to go!

### CODE REVIEW

File: manifests/scenarios/gmp-otel/1_collector.yaml [hunk 1]
```diff
@@ -190,34 +187,86 @@ data:
...
+          #              regex: (.+)
+          #            - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
+          #              action: replace
+          #              regex: (.+):(?:\d+);(\d+)
+          #              replacement: $$1:$$2
+          #              target_label: __address__
+          #            - action: labelmap
+          #              regex: __meta_kubernetes_pod_label_(.+)
 
-      prometheus/self-metrics:
```
Comment (bwplotka) [inline manifests/scenarios/gmp-otel/1_collector.yaml:213]: I removed this as I don't want to send this data to GMP, I want to scrape otel collector /metrics page. Do I need to configure something? 

I see some metrics like `otelcol_process_memory_rss` but no `otelcol_receiver_...` metrics. 🤔 Any ideas?

cc @ridwanmsharif

File: manifests/scenarios/gmp-otel/1_collector.yaml [hunk 2]
```diff
@@ -190,34 +187,86 @@ data:
       prometheus/bench:
         config:
           scrape_configs:
-          - job_name: otel-prom-bench
+          - job_name: PodMonitoring/gmp/avalanche/metrics
```
Comment (bwplotka) [inline manifests/scenarios/gmp-otel/1_collector.yaml:190]: Tried to do exactly match our relabelling, but looks like I broke it.. to be investigated. Any troubleshooting guidance for Otel, without Prometheus like target/service discovery UI?

File: tools/mtypes/main.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+			switch mtype {
+			case dto.MetricType_GAUGE:
+				fmt.Printf("--gauge-metric-count=%v\n", adjustedSeries)
+				adjustedSum += adjustedSeries
+			case dto.MetricType_COUNTER:
+				fmt.Printf("--counter-metric-count=%v\n", adjustedSeries)
+				adjustedSum += adjustedSeries
+			case dto.MetricType_HISTOGRAM:
+				avgBkts := s.buckets / s.series
+				adjustedSeries /= 2 + avgBkts
```
Comment (ridwanmsharif) [inline tools/mtypes/main.go:91]: Wouldn't this be:


```
adjustedSeries /= (2 + avgBkts + 1)
```

To account for the additional +inf bucket?

File: tools/mtypes/main.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+	writeStatistics(os.Stdout, total, statistics)
+
+	if *avalancheFlagsForTotal > 0 {
+		adjustedGoal := float64(*avalancheFlagsForTotal)
+		fmt.Println()
+		fmt.Println("Avalanche flags for the similar distribution to get to the adjusted series goal of:", adjustedGoal)
+
+		adjustedGoal = adjustedGoal / 10.0 // Assuming --series-count=10
+		adjustedSum := 0
+		for _, mtype := range allTypes {
```
Comment (ridwanmsharif) [inline tools/mtypes/main.go:76]: There is so much math here, I keep getting confused.

Maybe we can add a comment here to explain the algorithm? Or rename some of the vars to explain?

- `adjustedGoal` is tracking the # of series of that metric type that avalanche will have to create
- `adjustedSum` is tracking the total sum of series so far (at the end hopefully adjustedSum ~= adjustedGoal)
- `adjustedSeries` is tracking (per metric type), how many unique series of that metric type avalanche needs to create according to the ratio we got from our input example

If all this makes sense, then I think I have my head wrapped around all the math. Correct me if I got something wrong

File: tools/mtypes/main.go (hunk 3 above)
Comment (ridwanmsharif) [reply tools/mtypes/main.go:91]: Or is this more right:

```
avgBkts := ((s.buckets / s.series) - s.series) // get rid of the +inf bucket from each series
adjustedSeries /= 2 + avgBkts
```

The +inf bucket keeps throwing me off

**Review by ridwanmsharif — COMMENTED** [review]
only reviewed the `tools/mtypes` so far. Overall looks good but left a couple questions about the math. Will review the rest tommorrow

File: tools/mtypes/main.go (hunk 4 above)
Comment (bwplotka) [reply tools/mtypes/main.go:76]: Yup, we can add more comments, will add later. There is definitely room to rename metrics for clarity. Maybe `adjustedSeries` should be `adjustedSeriesForType`?

Comment (bwplotka) [reply tools/mtypes/main.go:76]: Otherwise your explanation is 100% right, nice!

File: manifests/scenarios/gmp-otel/1_collector.yaml [hunk 5]
```diff
@@ -169,55 +171,90 @@ data:
...
+      #          - delete_key(attributes, "namespace")
+      #          - set(attributes["exported_job"], attributes["job"])
+      #          - delete_key(attributes, "job")
+      #          - set(attributes["exported_instance"], attributes["instance"])
+      #          - delete_key(attributes, "instance")
+      #          - set(attributes["exported_project_id"], attributes["project_id"])
+      #          - delete_key(attributes, "project_id")
 
     receivers:
       prometheus/bench:
```
Comment (ridwanmsharif) [inline manifests/scenarios/gmp-otel/1_collector.yaml:192]: [nit] Lets add a comment here about why this scrape config is so complicated - it tries to replicate exactly what the operator creates

File: manifests/scenarios/gmp/1_collector.yaml [hunk 6]
```diff
@@ -172,9 +240,9 @@ spec:
         - name: config-out
           readOnly: true
           mountPath: /prometheus/config_out
-        - name: collection-secret
```
Comment (ridwanmsharif) [inline manifests/scenarios/gmp/1_collector.yaml:175]: [info] What was this for previously? Some operator thingy?

File: tools/mtypes/main.go (hunk 3 above)
Comment (ridwanmsharif) [reply tools/mtypes/main.go:91]: When I tried it for a spin it worked well so I trust the math, and its not important to be super accurate here anyway so SGTM

File: manifests/gmp-operator/1_setup.yaml [hunk 7]
```diff
@@ -1,3524 +0,0 @@
-# Copyright 2022 Google LLC
```
Comment (ridwanmsharif) [inline manifests/gmp-operator/1_setup.yaml:1]: I wonder if as a follow up its useful to have a manifest that does use the operator. Could be a useful way to see how operator versions affect the benchmarks over time (say if the operator adds some relabelling that proves to be very expensive).

File: .github/workflows/golangci-lint.yml [hunk 8]
```diff
@@ -1,31 +0,0 @@
----
-# This action is synced from https://github.com/prometheus/prometheus
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply .github/workflows/golangci-lint.yml:2]: Was Prometheus specific, giving some weird errors, let me bring full CI for lint, test at once. Also not sure if this tool placement is correct, could be a separate repo/promtool

File: manifests/gmp-operator/1_setup.yaml (hunk 7 above)
Comment (bwplotka) [reply manifests/gmp-operator/1_setup.yaml:1]: Yea, using operator blocks the architecture of this benchmark, but I see your point. It's actually quite easy to write a tool that will regenerate this scrape config based on operator code in prometheus-engine. Then we can hook it in CI/script, WDYT? Will add an issue

File: .github/workflows/golangci-lint.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (bwplotka) [reply .github/workflows/golangci-lint.yml:2]: Actually, let me do it here, let's do it properly. Promtool makes not that much sense, but in future we could move it to avalanche.

File: README.md [hunk 9]
```diff
@@ -1,16 +1,34 @@
...
+The general flow looks as follows:
+
+* You setup your GKE cluster once: `make cluster-setup CLUSTER_NAME=my-prombenchy`
+* Then to set up any benchmark run you do `make start CLUSTER_NAME=my-prombenchy BENCH_NAME=<name of benchmark, also k8s namespace> SCENARIO=./manifests/scenarios/gmp`. This will setup node-pool and your collector (e.g. as daemon set or separate pod - up to you, as long as you do correct node section!)
+
+You can start as many scenarios as you want on the single cluster (make sure to use unique `BENCH_NAME` though!)
+
+The scenario is a path to the "collector" manifest, so anything that will scrape `./manifests/load/avalanche.exampletarget.yaml`. Feel free to adjust anything in `./manifests/scenarios/` or add your own.
+
+This setup uses separate Prometheus for gathering metrics about core resources and collectors (available locally and in GCM). Make sure your pod has `app=collector` label and relevant port name has `-ins` suffix, sto be scraped by this core Prometheus. There is also a dashboard you can apply to GCM in `./dashboards/`.
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply README.md:21]: FYI: please use Suggestion feature instead (I can be more lazy then)  (: 

image

File: manifests/scenarios/gmp/1_collector.yaml (hunk 6 above)
Comment (bwplotka) [reply manifests/scenarios/gmp/1_collector.yaml:175]: Correct! It's actually empty if workload identity is used, so we can kill it.

File: manifests/gmp-operator/1_setup.yaml (hunk 7 above)
Comment (bwplotka) [reply manifests/gmp-operator/1_setup.yaml:1]: https://github.com/bwplotka/prombenchy/issues/13

**Pull request merged**